### ISSUE TITLE
Failing to load SVG file if url includes spaces

### ISSUE THREAD

**Issue by travisjayday** [Kozea/WeasyPrint#645]
This might be a problem with CairoSvg, not sure. 

When specifying a url in an html template for Weasyprint, the following works 

`<img src='file:///Users/name/Desktop/Select.svg' />`

while this 

`<img src='file:///Users/name/Desktop/foo%20bar/Select.svg' />`

fails with a weird error: 

`ERROR:weasyprint:Failed to draw an SVG image at file:///Users/name/Desktop/foo%20bar/Select.svg : <urlopen error [Errno 21] Is a directory: '/Volumes/.../.../.../'>`  (the last path in the message, ../../../ points to current working directory)

This is probably not expected behaviour and somewhat of an annoyance if directory names have spaces in them. With png images, however, it works just as expected, even with spaces. So something must be wrong with the .svg loading implementation.

**Comment by Tontyna** [comment]
If WeasyPrint cannot find the img it only informs you with `Failed to load image at "file:///...`.
The ERROR you see is the caught exception raised by cairosvg. So yes, it's Cairo.

Whats your Cairo version? My Cairo (version 1.15.10 ) has no problems loading svgs, neither with space nor with `%20` in the url.

**Comment by travisjayday** [comment]
Strange. I'm using CairoSvg version 2.1.3.

**Comment by Tontyna** [comment]
Mee too, have cairosvg version 2.1.3-.
But what I'm interested in is your **Cairo** version.

**Comment by travisjayday** [comment]
Sorry, I'm not very knowledgeable about Cairo :-/ 

```
    $ cairo-trace --version
    $ cairo-trace, version 1.14.12.
```

Is that how I check the version? If so, it appears I'm using an older version of Cairo, compared to yours... Maybe that's the problem.

**Comment by Tontyna** [comment]
Dont know what `cairo-trace` returns. To get the Cairo version WeasyPrint uses I run the following Python script

````python
import cairocffi as cairo

print('Cairo:', cairo.cairo_version_string ())
````

If your Cairo is v 1.14 then maybe updating it is a good idea, anyway.

Not having access to a Cairo 1.14  I can neither check whether Cairo is the root of the space-in-svg-path issue, nor try to find a workaround in `SVGImage.draw()`...

**Comment by Tontyna** [comment]
Hmm...  WeasyPrint  doesn't pass the SVGs url to cairosvg/Cairo at all. It passes on the content of the given SVG.
So I conclude Cairo only will try (and maybe fail) to load something when your SVG contains (**links to**) another file.

Question: Does your `Select.svg` ~~embed~~ link something ?

**Comment by Tontyna** [comment]
Sorry, I'm on Windows an know almost nothing about installing libraries on MacOS.

**Comment by liZe** [comment]
> Is there a better way of installing the latest Cairo (like through a package manager)?

WeasyPrint's documentation recommands Homebrew, but its current Cairo version is 1.14.12.

[Cairo 1.16.0 is coming soon](https://lists.cairographics.org/archives/cairo/2018-June/028628.html). As it will be a stable release, we can expect Homebrew to quickly provide a formula.

**Comment by liZe** [comment]
> [Cairo 1.16.0 is coming soon](https://lists.cairographics.org/archives/cairo/2018-June/028628.html). As it will be a stable release, we can expect Homebrew to quickly provide a formula.

Cairo 1.16.0 is available on Homebrew. @travisjayday Do you have the possibility to try this version?

**Comment by travisjayday** [comment]
Sorry, I cannot test it. I don't work at the company I used to so I don't
have access to a MacBook anymore.

**Comment by liZe** [comment]
Oh, no problem, I can actually reproduce the bug.

**Comment by liZe** [comment]
It's actually fixed in CairoSVG 2.3.0 (see Kozea/CairoSVG#226).
